# Técnicas de sumarización — Titanic

Asimetría, curtosis y tablas cruzadas.

In [1]:
import pandas as pd


url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

print(df.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [2]:
df["Age"].skew()

np.float64(0.38910778230082704)

### 1. Asimetría (Skewness) de `Age`

**Resultado: +0.389** — asimetría positiva.

**Qué significa:** la cola larga está del lado derecho, o sea del lado de las edades altas. Había más pasajeros jóvenes que ancianos: la mayoría se concentra entre los 20 y 35 años, y los pocos adultos mayores estiran la distribución hacia la derecha (solo 22 pasajeros pasaban de 60 años, contra 384 menores de 30).

Se confirma con media (29.70) > mediana (28.0): los mayores jalan el promedio hacia arriba. Al ser solo +0.39, la distribución es casi simétrica.

In [3]:
df["Fare"].kurt()

np.float64(33.39814088089868)

### 2. Curtosis (Kurtosis) de `Fare`

**Resultado: +33.40** — curtosis muy alta.

**Qué significa:** Un valor de 33 indica colas extremadamente gruesas, osea que hay muchísimos más valores lejos del centro de los que una normal predeciría, y al mismo tiempo un pico muy pronunciado de tarifas baratas.

Sí confirma los outliers. Hay los 116 pasajeros por encima de 65.63 y el caso extremo de 512.33 son precisamente lo que infla la curtosis a 33. Una distribución normal no produce un numero asi

In [4]:
pd.crosstab(df["Pclass"], df["Survived"])

Survived,0,1
Pclass,,
1,80,136
2,97,87
3,372,119


In [5]:
pd.crosstab(df["Pclass"], df["Survived"], normalize="index")

Survived,0,1
Pclass,,
1,0.370370,0.629630
2,0.527174,0.472826
3,0.757637,0.242363


### 3. Tabla cruzada (Cross-Tabulation) `Pclass` × `Survived`

| Pclass | Murió (0) | Sobrevivió (1) | % sobrevivió |
|---|---|---|---|
| 1ª | 80 | 136 | **62.96%** |
| 2ª | 97 | 87 | 47.28% |
| 3ª | 372 | 119 | **24.24%** |

**Patrón visible:** la supervivencia cae de forma monótona conforme baja la clase. La primera clase es el único grupo donde sobrevivieron más de los que murieron, en tercera murieron más de 3 por cada sobreviviente.

En números crudos el contraste es aún más fuerte, los 372 muertos de tercera clase son, por sí solos, más que todos los pasajeros de primera juntos. La clase socioeconómica funcionó como predictor directo de supervivencia.